infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterControllerV1.kt

```diff
@@ -1,13 +1,84 @@
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder
+import org.springframework.web.bind.annotation.ModelAttribute
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestParam
 
-@GeoviiteIntegrationApiController("/frame-converter/v1")
-class FrameConverterControllerV1 {
+@Configuration
+class FrameConverterApiObjectMapperV1 {
 
-    @GetMapping("/some-conversion")
-    fun someConversion(): String {
-        return "some response"
+    @Bean
+    fun objectMapper(builder: Jackson2ObjectMapperBuilder): ObjectMapper {
+        val module = SimpleModule()
+        module.addDeserializer(FrameConverterRequestV1::class.java, FrameConverterRequestDeserializerV1())
+        return builder.build<ObjectMapper>().registerModule(module)
+    }
+}
+
+@GeoviiteIntegrationApiController(
+    [
+        "/rata-vkm",
+        "/rata-vkm/dev",
+        "/rata-vkm/v1",
+        "/rata-vkm/dev/v1",
+    ],
+)
+class FrameConverterControllerV1 @Autowired constructor(
+    private val frameConverterServiceV1: FrameConverterServiceV1,
+) {
+
+    @RequestMapping(method = [RequestMethod.GET, RequestMethod.POST])
+    fun multiInputTransform(
+        @RequestParam(required = false, name = "json") requests: List<FrameConverterRequestV1>?,
+    ): GeoJsonFeatureCollection {
+        val parsedRequests = requests ?: emptyList()
+
+        return GeoJsonFeatureCollection(
+            features = parsedRequests.flatMap(::processRequest),
+        )
+    }
+
+    @RequestMapping(method = [RequestMethod.GET, RequestMethod.POST], params = ["!json"])
+    fun singleInputTransform(
+        @ModelAttribute request: CoordinateToTrackMeterRequestV1,
+    ): GeoJsonFeatureCollection {
+        return GeoJsonFeatureCollection(
+            features = processRequest(request),
+        )
+    }
+
+    private fun processRequest(request: FrameConverterRequestV1): List<GeoJsonFeature> {
+        return when (request) {
+            is CoordinateToTrackMeterRequestV1 -> {
+                val (validatedRequest, errorResponse)
+                    = frameConverterServiceV1.validateCoordinateToTrackMeterRequest(request)
+
+                if (validatedRequest == null) {
+                    errorResponse
+                } else {
+                    frameConverterServiceV1.coordinateToTrackAddress(
+                        layoutContext = MainLayoutContext.official,
+                        request = validatedRequest,
+                    )
+                }
+            }
+
+            else -> throw IntegrationApiException(
+                message = "Unsupported request type",
+                localizedMessageKey = "unsupported-request-type",
+            )
+        }
```

Comment:
Vaikuttaisi myös siltä, että täältä tulee nyt kahdenlaisia erilaisia virheitä ulos (osassa tulee GeoJSON-tyyppinen virhe (jota ei taideta mäpätä mihinkään ns. oikeaan HTTP-virheeseen, vaan josta palautunee OK 200?) ja toisissa heitetään suora HTTP-virheeksi mäppäytyvä poikkeus. Onko tämä speksattu ulkopuolelta tällaiseksi? Jotenkin kuulostaisi omaan korvaan oikeammalta jos a) kaikki virhetilanteet toimisivat samalla tavalla (olkoon virhevastaukset sitten joko meidän muuallakin käytetyssä virhemuodossa tai GeoJSON-virheinä) ja b) ihan kaikissa virhetilanteissa heitettäisiin eteenpäin joku oikea HTTP-virhe

Reply:
👍 Joo näin on, virheenkäsittelyä voisi ainakin vieläkin ehostaa standardimmaksi. Sitä ennen pitänee päättää halutaanko tosiaan se 1-1 suhde aiempien virheiden välille.